### PR TITLE
Refactor/cleanup

### PR DESCRIPTION
**Pull request by vyasr**
## Description
This branch has a bunch of miscellaneous cleanup tasks done:
* Removes Box.cc
* Removes all assertions
* Remove usage of using-declarations for std and tbb
* Removes outstanding usage of Index1D (added a couple necessary features to ManagedArray)

## Motivation and Context
Just some additional cleanup that's good to do and make further changes easier.

## How Has This Been Tested?
Existing tests pass.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [x] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

File: cpp/pmft/PMFTXYZ.cc [hunk 1]
```diff
@@ -78,7 +76,7 @@ void PMFTXYZ::accumulate(const locality::NeighborQuery* neighbor_query,
         for (unsigned int k = 0; k < n_faces; k++)
         {
             // create the extra quaternion
-            quat<float> qe(face_orientations[q_i(k, neighbor_bond.ref_id)]);
+            quat<float> qe(face_orientations[util::ManagedArray<unsigned int>::getIndex({neighbor_query->getNPoints(), n_faces}, {neighbor_bond.ref_id, k})]);
```
Comment (bdice) [inline cpp/pmft/PMFTXYZ.cc:79]: What this does is not clear to me at all. I would prefer to make this into two lines or add comments or something like that.

**Review by bdice — APPROVED** [review]
One change request, otherwise great!

Me reading this PR: https://www.youtube.com/watch?v=hm_yHwnySX8

File: cpp/pmft/PMFTXYZ.cc (hunk 1 above)
Comment (bdice) [reply cpp/pmft/PMFTXYZ.cc:79]: Alternatively make an indexing method that isn't `static`, using the shape of the instantiated object's underlying array?

Comment (vyasr) [reply cpp/pmft/PMFTXYZ.cc:79]: I'm not exactly sure what you're suggesting here. In the interest of moving forward, I made an issue to track this and I'm just going to resolve this and merge.

**Pull request merged**